### ISSUE TITLE
Incorrect signature types in compiled JsDoc annotations

### ISSUE THREAD

**Issue by mttrbrts** [accordproject/ergo#19]
Comparing original JS latedeliveryandpenalty sample with compiled JS from Ergo I see the following difference.

JS sample
```
/**
 * Execute the smart clause
 * @param {Context} context - the Accord context
 * @param {io.clause.latedeliveryandpenalty.LateDeliveryAndPenaltyRequest} context.request - the incoming request
 * @param {io.clause.latedeliveryandpenalty.LateDeliveryAndPenaltyResponse} context.response - the response
 * @AccordClauseLogic
 */
```

Compiled Jura
```
/**
 * Execute the smart clause
 * @param {Context} context - the Accord context
 * @param {Request} context.request - the incoming request
 * @param {Response} context.response - the response
 * @AccordClauseLogic
 */
```

I suspect that the missing namespaces are related to #18 , but the type names also seem to be wrong.

**Comment by jeromesimeon** [comment]
Our first bug report. Thanks!

**Comment by mttrbrts** [comment]
Thanks Jerome, once fixed, these test cases in cicero-core will need to amended.

https://github.com/accordproject/cicero/blob/f9c6cf6af6fe678ebca9344dffd66bb59cfc9cc7/packages/cicero-core/test/template.js#L149
https://github.com/accordproject/cicero/blob/f9c6cf6af6fe678ebca9344dffd66bb59cfc9cc7/packages/cicero-core/test/template.js#L169

**Comment by jeromesimeon** [comment]
Just to bounce back on this issue. I think this should be reviewed with whole-contract support in mind. It's unclear we should keep dispatching in Cicero the way it is currently done when we have full contracts. If dispatching isn't needed, the issue might be simpler or actually become moot.

**Comment by jeromesimeon** [comment]
Fixed in c8a325532d331292792e432a057016b25d7e53f3. Also fixed the corresponding Cicero-core tests.
